### DISCUSSION TITLE
Postgres Conductor queue_message query taking 100% cpu during peak load

### DISCUSSION THREAD

**Discussion by shekarpcs** [Netflix/conductor#3561] [General]
Recently we divided main workflow into subworkflows. As soon as this changes in deployed production following query taking long time and consuming 100% of cpu on db. 

SELECT message_id, priority, payload FROM queue_message WHERE queue_name = $1 AND popped = ? AND deliver_on <= (current_timestamp + (? ||?)::interval)
ORDER BY priority DESC, deliver_on, created_on LIMIT $2 FOR UPDATE SKIP LOCKED"

Db is postgres, 
queue implementation postgres
conducton : 3.13.2 

As it is prod issue, we have to figure out how to reduce this. is throwing more capacity only the option now?

**Comment by Kiranbo97** [comment]
Hi
have successfully establish connection between postgres DB with out any error
